### ISSUE TITLE
wp_ppprotect table empty causing foreach error

### ISSUE THREAD

**Issue by bretterer** [Ontraport/PilotPress#45]
On lines https://github.com/Ontraport/PilotPress/blob/master/ppprotect-categories.php#L345-L351 of the code, the method is expecting an array to be passed into the foreach loop.  Looking into the database, and what the method does, the table `wp_ppprotect` is empty on my site causing:

```
Notice: Trying to get property of non-object in /wp-content/plugins/pilotpress/ppprotect-categories.php on line 344

Warning: Invalid argument supplied for foreach() in /wp-content/plugins/pilotpress/ppprotect-categories.php on line 345

Notice: Trying to get property of non-object in /wp-content/plugins/pilotpress/ppprotect-categories.php on line 358
```

This is a major issue as it halts the rendering of PHP.

**Comment by willdeangelis** [comment]
Hi there,

Sorry about this. We will be releasing a new version tomorrow with the fixes once our QA team signs off on it. It will be 1.9.1. I will write you back here once it has been released.

Best!

william

**Comment by bretterer** [comment]
That is great to hear @willdeangelis   Thanks and I look forward to seeing if this fixes my issues.

-Brian

**Comment by bretterer** [comment]
I just did the update to our site and I am still having issues..


```
Notice: Trying to get property of non-object in /wp-content/plugins/pilotpress/ppprotect-categories.php on line 470

Notice: Trying to get property of non-object in /wp-content/plugins/pilotpress/ppprotect-categories.php on line 487

Notice: Trying to get property of non-object in /wp-content/plugins/pilotpress/ppprotect-categories.php on line 487

Notice: Trying to get property of non-object in /wp-content/plugins/pilotpress/ppprotect-categories.php on line 493
```

**Comment by willdeangelis** [comment]
Hi again... sorry to hear that. Unfortunately for some reason I'm unable to replicate your errors. That said based upon the log entries you sent I made a change and am hoping you can test it for me. Can you open up the attached zip file and replace the current ppprotect-categories.php file in your current install with the one in the zip? Let me know.

Many thanks!

w

[ppprotect-categories.zip](https://github.com/Ontraport/PilotPress/files/520120/ppprotect-categories.zip)

**Comment by bretterer** [comment]
@willdeangelis That fixed a major part of the issue.  Now the only thing that is odd is if I have Yoast SEO AND PilotPress enabled, my sidebar's do not show on my blog overview page.

**Comment by willdeangelis** [comment]
@bretterer - That's odd. Having issues replicating that one too. I have the following setup:

- WordPress 4.6.1 (Latest)
- Yoast SEO (Latest)
- PilotPress (Latest)
- Error reporting: E_ALL
- No category perms set
- No individual post perms set on the posts in the category I'm checking

Sidebar is working for me and getting no errors in the error log - http://www.awesomescreenshot.com/image/1694284/021f0a970f4f24db3db496b42698f718

Then I tested adding perms to the categories, sidebar is still displaying and getting no errors in the error log.

Then I tested adding individual perms, the sidebar is still displaying and getting no errors in the error log.

 Are you getting any errors or notices that might help me track this down?

**Comment by bretterer** [comment]
Can I send you a backup buddy backup of the site so you can see what I am seeing?

**Comment by willdeangelis** [comment]
We unfortunately don't do that when supporting this plugin because of possible security risks with foreign code on our machines. If you have error log entries, screenshots or screencasts I'm more than happy to track things down that way. Unfortunately without anything like that it's difficult for me to troubleshoot.

Here are a few questions that may help:

- If you deactivate Yoast does everything work ok? 
- If you re-activate Yoast and deactivate PilotPress does everything work ok?
- If you change your theme to one of the WordPress defaults does everything work ok?
- Just for clarification... you are talking about the sidebar on the front end of the side... not in the admin area right?

Let me know. Best!

w

**Comment by bretterer** [comment]
@willdeangelis We found the remaining bug here.  Thank you, and would you explain what the fix you did was so I can understand what was going on?  Also, will this be pushed as 1.9.2?

**Comment by willdeangelis** [comment]
Ok great. Good to hear.

Basically the notices you had told me that you had an empty permissions table and as such the object I was trying to use was not actually an object. It was empty hence it was generating the errors you had. I simply added a check to see if the object is empty or not and if this is the case then I prevented it from executing the code. This is very similar to the fixes in 1.9.1.

And yes... this will be posted as 1.9.2 later on today.

Best!

w
